include/hip/hcc_detail/hip_runtime.h

```diff
@@ -389,11 +389,26 @@ extern void ihipPostLaunchKernel(const char* kernelName, hipStream_t stream, gri
 
 typedef int hipLaunchParm;
 
+#ifdef HIP_MACRO_BASED_GGL
+
 #define hipLaunchKernelGGL(kernelName, numblocks, numthreads, memperblock, streamId, ...)          \
     do {                                                                                           \
         kernelName<<<(numblocks), (numthreads), (memperblock), (streamId)>>>(__VA_ARGS__);         \
     } while (0)
 
+#else
```

Comment:
Why we need this then?

Reply:
I guess this is for debugging purpose in case issues arise. After it passes all math libs and frameworks we may remove this.

Reply:
yes, will remove the macro in the future